src/structures/TextChannel.js

```diff
@@ -48,6 +49,17 @@ class TextChannel extends GuildChannel {
     return this.edit({ nsfw }, reason);
   }
 
+  /**
+   * Sets the category parent of this channel.
+   * @param {TextChannel|VoiceChannel|Snowflake} channel Parent channel
+   * @param {boolean} [options.lockPermissions] Lock the permissions to what the parent's permissions are
+   * @param {string} [options.reason] Reason for modifying the parent of this channel
+   * @returns {Promise<TextChannel>}
+   */
+  setParent(channel, { lockPermissions = true, reason } = {}) {
```

Comment:
You could bind the method itself in the constructor with something like `this.setParent = setParent.bind(this)` (or maybe Reflect.defineProperty because it's a method)
And document it with `@instance` and `@name` like it's done for `GuildStore#resolve`